src/wolfio.c

```diff
@@ -2182,20 +2182,24 @@ int MicriumReceiveFrom(WOLFSSL *ssl, char *buf, int sz, void *ctx)
     NET_SOCK_ADDR_LEN peerSz = sizeof(peer);
     NET_SOCK_RTN_CODE ret;
     NET_ERR err;
-    int dtls_timeout = wolfSSL_dtls_get_current_timeout(ssl);
 
     WOLFSSL_ENTER("MicriumReceiveFrom()");
 
-    if (ssl->options.handShakeDone)
-        dtls_timeout = 0;
+#ifdef WOLFSSL_DTLS
+    {
+        int dtls_timeout = wolfSSL_dtls_get_current_timeout(ssl);
+        if (ssl->options.handShakeDone)
+            dtls_timeout = 0;
```

Comment:
Also, the value of the `dtls_timeout` should be adjusted like it is adjusted in `EmbedReceiveFrom`
```           
 #ifdef WOLFSSL_DTLS13
            if (wolfSSL_dtls13_use_quick_timeout(ssl) &&
                IsAtLeastTLSv1_3(ssl->version))
                timeout /= 4;
            #endif /* WOLFSSL_DTLS13 */
        #else
            struct timeval timeout;
            XMEMSET(&timeout, 0, sizeof(timeout));
            #ifdef WOLFSSL_DTLS13
            if (wolfSSL_dtls13_use_quick_timeout(ssl) &&
                IsAtLeastTLSv1_3(ssl->version)) {
                if (dtls_timeout >= 4)
                    timeout.tv_sec = dtls_timeout / 4;
                else
                    timeout.tv_usec = dtls_timeout * 1000000 / 4;
            }
            else
            #endif /* WOLFSSL_DTLS13 */
```

Reply:
@rizlik , 

Thats odd .  in `EmbedReceiveFrom()` the call to `wolfSSL_dtls_get_current_timeout()` is not guarded by `WOLFSSL_DTLS`

Reply:
Oh....the entire function is guarded outside by `WOLFSSL_DTLS`

Reply:
The entire `EmbedReceiveFrom()` is included in a `WOLFSSL_DTLS` guard (together with other functions).

Reply:
Note that I will do my best to make a fix that looks right...but I'm weary of doing this because I cannot test it and might introduce some sort of less visible bug or vulnerability.